clippy_lints/src/strings.rs

```diff
@@ -438,27 +440,94 @@ declare_clippy_lint! {
 
 declare_lint_pass!(StringToString => [STRING_TO_STRING]);
 
+fn is_parent_map_like(cx: &LateContext<'_>, expr: &Expr<'_>) -> Option<rustc_span::Span> {
+    if let Some(parent_expr) = get_parent_expr(cx, expr)
+        && let ExprKind::MethodCall(name, _, _, parent_span) = parent_expr.kind
+        && name.ident.name == sym::map
+        && let Some(caller_def_id) = cx.typeck_results().type_dependent_def_id(parent_expr.hir_id)
+        && (clippy_utils::is_diag_item_method(cx, caller_def_id, sym::Result)
+            || clippy_utils::is_diag_item_method(cx, caller_def_id, sym::Option)
+            || clippy_utils::is_diag_trait_item(cx, caller_def_id, sym::Iterator))
+    {
+        Some(parent_span)
+    } else {
+        None
+    }
+}
+
+fn is_called_from_map_like(cx: &LateContext<'_>, expr: &Expr<'_>) -> Option<rustc_span::Span> {
+    // Look for a closure as parent of `expr`, discarding simple blocks
+    let parent_closure = cx
```

Comment:
We have a `clippy_utils::peel_blocks` function. Could this be used here instead?

Reply:
It's not doing the same thing so it doesn't work here.

Reply:
Ah right, it works in the opposite direction.

Reply:
Yes, and as I pointed out in <https://github.com/rust-lang/rust-clippy/pull/14396#discussion_r1992198450>, this would be rather dangerous to put this expression in a utility function, as it may not do what users expect when blocks are "indirect" part of an expression (such as the "else" block of an "if" expression).